packages/rainbowkit/src/config/getDefaultConfig.ts

```diff
@@ -1,68 +1,93 @@
-import { type Chain, configureChains, createConfig } from 'wagmi';
-import { publicProvider } from 'wagmi/providers/public';
+import { Transport } from 'viem';
+import { http } from 'wagmi';
+import { WagmiProviderProps, createConfig } from 'wagmi';
+import { type Chain } from 'wagmi/chains';
 import type { WalletList } from '../wallets/Wallet';
+import { computeWalletConnectMetaData } from '../wallets/computeWalletConnectMetaData';
 import { connectorsForWallets } from '../wallets/connectorsForWallets';
-import type { WalletConnectConnectorMetadata } from '../wallets/getWalletConnectConnector';
 import {
   coinbaseWallet,
   metaMaskWallet,
   rainbowWallet,
   walletConnectWallet,
 } from '../wallets/walletConnectors';
 
-export const getDefaultConfig = ({
-  appName,
-  appDescription,
-  appUrl,
-  appIcon,
-  chains,
-  wallets,
-  projectId,
-}: {
+export type _chains = readonly [Chain, ...Chain[]];
+
+// Define the '_transports' type as a Record
+// It maps each 'Chain' id to a 'Transport'
+export type _transports = Record<_chains[number]['id'], Transport>;
+
+interface GetDefaultConfigParameters {
   appName: string;
   appDescription?: string;
   appUrl?: string;
   appIcon?: string;
-  chains: Chain[];
   wallets?: WalletList;
   projectId: string;
-}) => {
-  const { publicClient, webSocketPublicClient } = configureChains(chains, [
-    publicProvider(),
-  ]);
+  chains: _chains;
+  transports?: _transports;
+  multiInjectedProviderDiscovery?: boolean;
+}
+
+const createDefaultTransports = (chains: _chains): _transports => {
+  const transportsObject = chains.reduce((acc: _transports, chain) => {
+    const key = chain.id as keyof _transports;
+    acc[key] = http() as _transports[keyof _transports]; // Type assertion here
+    return acc;
+  }, {} as _transports);
 
-  const metadata: WalletConnectConnectorMetadata = {
-    name: appName,
-    description: appDescription,
-    url: appUrl,
-  };
+  return transportsObject;
+};
+
+export const getDefaultConfig = ({
+  appName,
+  appDescription,
+  appUrl,
+  appIcon,
+  wallets,
+  projectId,
+  chains,
+  multiInjectedProviderDiscovery = true,
+  transports,
+}: GetDefaultConfigParameters): WagmiProviderProps['config'] => {
```

Comment:
Is `WagmiProviderProps['config']` the right return type here?

Reply:
Yeah. You can see it also in `<WagmiProvider />`
![image](https://github.com/rainbow-me/rainbowkit/assets/53529533/607a6acb-d0ed-4d63-b74e-d9f8902c4684)